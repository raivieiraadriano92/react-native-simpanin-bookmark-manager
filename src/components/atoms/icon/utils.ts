import normalize from 'react-native-normalize'

type ResizeProps = {
  height: number
  size: number
  width: number
}

type ResizeReturn = Pick<ResizeProps, 'height' | 'width'>

export function resize({ height, size, width }: ResizeProps): ResizeReturn {
  let newHeight
  let newWidth

  if (height === width) {
    newHeight = size
    newWidth = size
  } else if (height > width) {
    newHeight = size
    newWidth = (size * width) / height
  } else {
    newHeight = (size * height) / width
    newWidth = size
  }

  return {
    height: normalize(newHeight),
    width: normalize(newWidth)
  }
}
