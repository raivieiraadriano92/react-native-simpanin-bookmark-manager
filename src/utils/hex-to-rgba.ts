import Color from 'color'

export default function (hex: string, alpha = 1): string {
  return `rgba(${Color(hex).rgb().array().join(',')},${alpha})`
}
