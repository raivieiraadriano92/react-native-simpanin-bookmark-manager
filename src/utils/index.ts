export { default as hexToRgba } from './hex-to-rgba'
