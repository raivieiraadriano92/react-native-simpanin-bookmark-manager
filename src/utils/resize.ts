import normalize from 'react-native-normalize'

type Props = {
  base?: 'height' | 'width'
  height: number
  size: number
  width: number
}

export default function ({
  base,
  height,
  size,
  width
}: Props): Pick<Props, 'height' | 'width'> {
  let newHeight = size
  let newWidth = size

  if (height > width || base === 'height') {
    newHeight = size
    newWidth = (size * width) / height
  } else if (height < width || base === 'width') {
    newHeight = (size * height) / width
    newWidth = size
  }

  return {
    height: normalize(newHeight),
    width: normalize(newWidth)
  }
}
