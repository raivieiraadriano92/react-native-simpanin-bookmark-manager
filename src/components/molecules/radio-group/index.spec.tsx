import React from 'react'

import { render } from 'test/utils'

import RadioGroup from './index'

describe('<RadioGroup />', () => {
  it('should render correctly', () => {
    const { toJSON, unmount } = render(
      <RadioGroup
        onChange={() => {
          //
        }}
        options={[{ label: 'test', value: 0 }]}
        value={0}
      />
    )

    expect(toJSON()).toMatchSnapshot()

    unmount()
  })
})
