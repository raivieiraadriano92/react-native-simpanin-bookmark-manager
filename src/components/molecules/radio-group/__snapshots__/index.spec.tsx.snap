// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioGroup /> should render correctly 1`] = `
<RNCSafeAreaProvider
  onInsetsChange={[Function]}
  style={
    Array [
      Object {
        "flex": 1,
      },
      undefined,
    ]
  }
/>
`;
